packages/stream-api/src/streamApi.ts

```diff
@@ -54,6 +68,8 @@ export class StreamApi {
       ...descriptor,
       url: appendSettingsToUrl(descriptor.url, settings),
     };
+    window.addEventListener('offline', this.handleOffline);
+    window.addEventListener('online', this.handleOnline);
```

Comment:
I'm not sure if this is the right place to put this logic. I'd like to leave open the possibility of using this client in a Node environment, which is possible now by passing a Node compatible `WebSocketClient`. I'd consider maybe moving this into some helper class/module in the viewer package, OR using the decorator pattern to wrap this client to provide this additional behavior.